addons/web/models/models.py

```diff
@@ -616,9 +634,20 @@ def _web_read_group_format(
         for dict_group in result:
             dict_group['__extra_domain'] = AND(dict_group.pop('__extra_domains'))
 
+        # Link the monetary columns to their currency columm
+        currency_column_mapping = {}
+        column_values = list(zip(*groups))[len(groupby):]
+        for agg in aggregates:
+            if currency_field_agg := self._get_currency_field_agg(agg):
+                currency_column_mapping[agg] = column_values[aggregates.index(currency_field_agg)]
```

Comment:
Instead of mapping only the currency field aggregates, why not mapping all aggregates to their columns.
It means that `index` become unnecessary and then the `list(zip(*groups))` too.